### PR TITLE
test: close concurrency and authority coverage gaps across store/server/workflows

### PR DESCRIPTION
**Pull request by oxoxDev**
## Summary

Closes several coverage gaps in edge-case handling found while auditing this codebase's unhappy paths — each test below asserts a behavior that was previously either implemented but unverified, or explicitly documented as a known limitation but never pinned. One unrelated one-line fix is bundled (see below) because it was blocking the test build on this base.

- **`write_atomic` under real concurrency, with no lock.** `path_lock`'s own doc comment says cross-process safety over one bundle rests entirely on `write_atomic` bounding damage to a lost update and never a torn file — a claim nothing exercised under actual concurrent writers. Added a test that spawns many concurrent `write_atomic_bytes` calls against one path, bypassing the in-process lock the way two separate `opencompany` processes would, and asserts the file that lands is always exactly one writer's full payload, never a mixed or truncated one.
- **Company rename is admin-only, but only by convention.** `PATCH {scope}` on the company profile calls `require_admin` manually in the handler body rather than through an extractor, and no test in that module ever drove a plain member against it. Added one.
- **Inference runtime restart is admin-only, but only by extractor, never proven.** `POST …/inference/restart` takes `AdminScopedCompany`, but every existing test in that module authenticates as the seeded admin. Added a member-refused test that also confirms a refused restart never swaps the runtime.
- **SMTP test-send's other refusal.** The existing test covers "no sender wired" (404). Nothing covered "sender wired, but the company never stored credentials" (400, `no SMTP credentials configured`) — a different, more common misconfiguration. Added it, and confirmed a refused send never reaches the injected sender.
- **Two racing legacy-home migrations.** The migration module's own docs describe `serve` booting while a hand-run `export` migrates the same home concurrently, and its existing test for that tolerance stages the *aftermath* of a race by hand rather than driving one. Added a real two-thread, barrier-synchronized race over one migration and asserted it never loses, duplicates, or corrupts the bundle.
- **A documented dry-run/live-guard divergence, previously unpinned.** The workflow dry-run's HTTP preflight check has a doc comment describing a deliberate choice: when a company's URL allowlist is entirely unparseable, the real guard fails closed (refuses every request) while the dry-run stub leaves it unchecked (reports success). That means a workflow with a badly misconfigured allowlist previews clean and then refuses every live request — a real, if narrow, false-assurance gap. Added a test that drives one request through both the real guard and the dry-run stub and pins the documented divergence, so a future change to either side has to touch it on purpose.

**Bundled, separate commit:** `fix(graphql): restore overlay_desk_hive in the two-company bridge fixture` — a pre-existing, one-line breakage on this base (a struct-literal fixture missed a field added by an earlier merge) that fails `cargo test --lib` outright. Included only because none of the tests above could be compiled or run without it; unrelated to the audit work otherwise.

## API Or Behavior Changes

None. Test-only, plus the one-line pre-existing test-fixture fix described above.

## Tests

This branch is forked from an `upstream/main` whose `Cargo.lock` pins `vendor/openhuman` at `0.63.24` while the submodule itself is checked out at `0.63.23` — `cargo … --locked` cannot resolve, and any invocation without `--locked` silently rewrites the lock file down to `0.63.23` (visible as an unwanted `Cargo.lock` diff on every local run). **`Cargo.lock` is untouched by this branch on purpose** — this base needs a rebase once that pin is fixed upstream, independent of this PR. Tests below were run with `--locked` dropped for that reason.

For each test: the production mechanism it names was broken, the test was confirmed to fail **on its own assertion** (not a compile error, not an unrelated panic), then the code was restored and the test reconfirmed green. Verified after all six break/restore cycles that `git diff | grep -E "^-" | grep -vE "^---" | grep -vE "^-\s*(//|///)"` is empty — no unrestored break left in production code.

1. **`store::fs::test::concurrent_writers_without_the_lock_never_leave_a_torn_file`** — broke `write_atomic_bytes` to skip the tmp-file-plus-rename dance in favor of a naive truncate-then-stream straight to the destination path, writing in 4 KiB chunks with a yield between each (the literal shape the function's own doc comment warns a plain `tokio::fs::write` degrades to). Failure:
   ```
   thread 'store::fs::test::concurrent_writers_without_the_lock_never_leave_a_torn_file' panicked at src/store/fs.rs:3747:9:
   the file's bytes were not a single writer's payload in full — a torn or interleaved write slipped through the lock-free path
   ```
   Note: a first attempt just swapping in a single non-chunked `tokio::fs::write` call did **not** turn the test red — a lone 200 KiB `write()` syscall lands atomically on this filesystem even with no tmp+rename, so that break proved nothing and was discarded before landing on the chunked version above, which does expose real interleaving.

2. **`server::ops::company_profile::tests::a_member_is_refused`** — commented out the `require_admin(...)` call in `patch_company`. Failure:
   ```
   thread 'server::ops::company_profile::tests::a_member_is_refused' panicked at src/server/ops/company_profile.rs:395:9:
   assertion `left == right` failed
     left: 200
    right: 403
   ```

3. **`server::ops::inference::tests::a_member_may_not_restart_the_runtime`** — disabled the role check inside `require_admin` (`if false && !principal.may_administer()`), which is what `AdminScopedCompany`'s extractor calls for `restart_runtime`. Failure:
   ```
   thread 'server::ops::inference::tests::a_member_may_not_restart_the_runtime' panicked at src/server/ops/inference.rs:1329:9:
   assertion `left == right` failed
     left: 200
    right: 403
   ```

4. **`server::ops::test::smtp_test_with_a_sender_but_no_stored_credentials_is_400`** — replaced the "no credentials" refusal in `run_test` with a fabricated empty `SmtpCredentials` fallback so it proceeds instead of refusing. Failure:
   ```
   thread 'server::ops::test::smtp_test_with_a_sender_but_no_stored_credentials_is_400' panicked at src/server/ops/test.rs:274:5:
   assertion `left == right` failed
     left: 200
    right: 400
   ```

5. **`store::migrate::test::two_concurrent_migrations_of_the_same_home_never_lose_or_duplicate_a_bundle`** — removed the `&& exists(to)` tolerance from `rename_or_already_moved`, so the losing thread's `NotFound` propagates as a hard error instead of being read as "already moved". Failure:
   ```
   thread 'store::migrate::test::two_concurrent_migrations_of_the_same_home_never_lose_or_duplicate_a_bundle' panicked at src/store/migrate.rs:1182:22:
   the losing thread must not abort with the winner's NotFound: StoreIo { path: "…/companies/companies/acme", source: Os { code: 2, kind: NotFound, message: "No such file or directory" } }
   ```

6. **`workflows::caps::dry_run::tests::dry_http_passes_an_entirely_malformed_allowlist_the_real_guard_fails_closed_on`** (`--features openhuman`) — **red proof not completed.** This feature's build compiles the full vendored `openhuman` crate graph from scratch and, with several other agents building on this host concurrently, did not finish within the time available. Not listing this as a vague "owed" item: it is compiled-clean (confirmed as part of the same `--lib --features openhuman` build that produced no errors) but neither its green baseline nor its red proof have a recorded run. Will follow up in this PR thread with both once a build completes; it is not a concurrency test, so there is no structural reason it can't be red-proofed the same way as the other five — it's purely a compile-time budget problem on this host, not attempted-and-flaky.

All five default-feature tests confirmed green again after restore:
```
test store::fs::test::concurrent_writers_without_the_lock_never_leave_a_torn_file ... ok
test store::migrate::test::two_concurrent_migrations_of_the_same_home_never_lose_or_duplicate_a_bundle ... ok
test server::ops::test::smtp_test_with_a_sender_but_no_stored_credentials_is_400 ... ok
test server::ops::company_profile::tests::a_member_is_refused ... ok
test server::ops::inference::tests::a_member_may_not_restart_the_runtime ... ok
```

- [ ] `cargo fmt --all -- --check` — not run locally
- [ ] `cargo clippy --all-targets --no-deps -- -D warnings` — not run locally
- [ ] `cargo build --all-targets` — not run locally
- [x] `cargo test --lib <targeted>` — 5 of 6 confirmed passing with red proofs (see above); 6th compiled, red proof pending

## Documentation

None needed — these are internal test additions and one test-fixture fix; no public behavior or docs changed.

## Summary by CodeRabbit

- **Tests**
  - Expanded permission coverage to ensure members cannot change company details or restart protected services.
  - Added validation for SMTP test sends when credentials are unavailable.
  - Strengthened concurrent storage tests to prevent corrupted or incomplete files.
  - Repeated migration tests to verify safe, duplicate-free results during concurrent operations.
  - Verified that dry-run request filtering matches live behavior when allowlists contain invalid entries.

### CODE REVIEW

File: src/store/fs.rs [hunk 1]
```diff
@@ -3676,6 +3676,63 @@ mod test {
...
+        // truncation.
+        const BYTES: usize = 200 * 1024;
+        let candidates: Vec<Vec<u8>> = (0..WRITERS)
+            .map(|writer| vec![b'A' + writer; BYTES])
+            .collect();
+
+        let mut set = tokio::task::JoinSet::new();
+        for candidate in candidates.clone() {
+            let path = path.clone();
+            set.spawn(async move { write_atomic_bytes(&path, &candidate).await });
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply src/store/fs.rs]: Right — fixed in fad200f10. `JoinSet::spawn` only queues the tasks, so a serial schedule passed this without ever reaching the contended path. All twelve writers now wait on a `tokio::sync::Barrier` released immediately before `write_atomic_bytes`. Worth noting the original red proof did expose a torn write, but only once the break was a chunked yielding stream — a single 200KB `write()` lands atomically on this filesystem regardless, so the test was already narrower than it looked.

File: src/store/migrate.rs [hunk 2]
```diff
@@ -1145,6 +1145,65 @@ mod test {
...
+        home.write(
+            "companies/companies/acme/company.toml",
+            "[company]\nname = \"Acme\"\n",
+        );
+
+        let barrier = std::sync::Arc::new(std::sync::Barrier::new(2));
+        let path_a = home.path().to_path_buf();
+        let barrier_a = barrier.clone();
+        let handle_a = std::thread::spawn(move || {
+            barrier_a.wait();
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply src/store/migrate.rs]: Correct, and the sharper half of the finding: the barrier releases before `migrate_legacy_nest`, which scans before it renames, so a winner finishing before the loser scans never reaches the tolerated `NotFound` at all. Rather than add a production seam purely for testability, fad200f10 repeats the race 24 times with a fresh home each attempt, asserting the same invariants every pass — the interleaving becomes near-certain instead of lucky, and a regression fails on whichever attempt exposes it. If you would still prefer an injected renamer, say so and I will add one.

[the remark this answers is not in the document]
Comment (oxoxDev) [reply src/store/migrate.rs]: You were right, and it was worse than a weak test.

I did not add a renamer hook — instead the new sibling test barriers directly at `rename_or_already_moved`, so nothing sits between the barrier release and the syscall and the contended path is the only path it can take.

Making the interleaving actually happen turned up two real defects, both now fixed in this branch:

1. A loser reaching the destination check after the winner's rename recorded a spurious `Collision` — "two bundles want this name" — for a migration that had just succeeded. A destination that appeared while the legacy entry vanished is that other process's move, not contention.
2. A loser scanning the nest while the winner removed it took a `NotFound` from mid-iteration and aborted the whole migration — exactly the boot failure `rename_or_already_moved`'s tolerance exists to prevent. `read_dir_names` already treated an absent directory as empty; a directory that disappears mid-scan is the same answer.

Measured on the whole-migration race test: 5 failures in 12 runs before, 1 in 12 after the first fix, 0 in 20 after both.

**Comment by oxoxDev** [comment]
@coderabbitai review
